admin/deployments.go

```diff
@@ -90,7 +92,18 @@ func (s *Service) StopDeployment(ctx context.Context, depl *database.Deployment)
 	return nil
 }
 
-func (s *Service) UpdateDeployment(ctx context.Context, depl *database.Deployment) error {
+func (s *Service) UpdateDeployment(ctx context.Context, depl *database.Deployment, opts *database.UpdateDeploymentOptions) error {
+	// Update the deployment if deployment parameters have changed
+	updateDeployment := opts.Branch != depl.Branch || opts.RuntimeHost != depl.RuntimeHost ||
+		opts.RuntimeInstanceID != depl.RuntimeInstanceID ||
+		opts.RuntimeAudience != depl.RuntimeAudience
+	if updateDeployment {
+		_, err := s.DB.UpdateDeployment(ctx, depl.ID, opts)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
I'm not sure this is safe to call from outside the deployment reconcile job. Particularly the changing of the runtime settings. @kaspersjo how would you approach this?